microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/HelidonProxyServices.java

```diff
@@ -75,8 +76,17 @@ public boolean supportsClassDefining() {
             // classloader, preventing it from seeing these fields/methods
             return defineClassSamePackage(originalClass, className, classBytes, off, len);
         } else {
-            // use a custom classloader to define classes in a new package
-            return wrapCl(originalClass.getClassLoader()).doDefineClass(originalClass, className, classBytes, off, len);
+            // try to use same package approach (maybe in same module?) to support package local methods
+            try {
+                return defineClassSamePackage(originalClass, className, classBytes, off, len);
```

Comment:
Maybe refactor so this statement happens only once instead of twice, so something like:
```
try {
    return defineClassSamePackage()...
} catch (Exception e) {
    if (samePackage(originalClass, className)) {
        throw e or wrapped e;
    } else {
        // carry on
    }
}
```

Reply:
Fixed